tests/fuzzer.c

```diff
@@ -1999,14 +1999,14 @@ static int basicUnitTests(U32 const seed, double compressibility)
 
     /* long rle test */
     {   size_t sampleSize = 0;
+        size_t expectedCompressedSize = 39; /* two RLE blocks, zstd 1.4.4 */
```

Comment:
minor, but needs to be correctly reflected in comments :

The below test will actually produce one compressed block, and then one RLE block.

The reason is : by default, and especially at level 1, the algorithm makes no educated choice regarding block boundaries, and therefore always cut at 128 KB boundaries.

As a consequence, if I understand the test correctly, it will generate a first block with 96 KB of 'A' and 32 KB of 'B', which is not RLE. Then the second block will be `128 KB - 1` of 'B', and this one should be RLE.

That's still good enough for tests, it's just that the code comment should accurately reflects the outcome.
Also, manually check that, when the algorithm _doesn't_ produce RLE blocks, then the test correctly fires up.

P.S : Selecting block boundaries for optimal compression ratio could be an interesting follow up for high compression modes. It's complex though, so drop me a word if you feel interested.

Reply:
Ah, good to know! I mistakenly assumed some boundary selection was already happening.

In that case, since we consume 96KB of 'A' and 256KB of 'B' in 128KB block increments, then it seems like there are three blocks right? First is 96KB 'A' + 32KB 'B', second is 128KB 'B', and 3rd is 96KB 'B', right?

Reply:
Sorry, I misread the code.
I though that the second `memset(,'A',)` was overwriting the first 96 KB of `'B'.
But since `sampleSize` is updated, it actually follows, so it's a different outcome.

I would predict 2 RLE blocks, the 1st and the 3rd.
Without intelligent boundary selection, the 2nd block cannot become RLE, because of the presence of a lone 'A' at the end of the block.

Still, good enough and appropriate for an RLE test.

Reply:
Yep you're right, I just printed out the blocks to be sure. I've updated the comment again
```
test126 : Long RLE test : 
RLE
notRLE
RLE
```